File: atcoder.jp/contests/abc176/tasks/abc176_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const sum = (array: number[]) => array.reduce((a, b) => a + b, 0);

const main = () => {
  const N = input();

  const ans = sum(N.split("").map(Number)) % 9 === 0;
  console.log(ans ? "Yes" : "No");
};

main();

File: atcoder.jp/contests/abc176/tasks/abc176_b/main.hs
import Data.Char (digitToInt)

yesNo :: Bool -> String
yesNo False = "No"
yesNo True = "Yes"

main :: IO ()
main = do
  n <- getLine

  let ans = sum (map digitToInt n) `mod` 9 == 0
  putStrLn $ yesNo ans
